electron/libelectron/src/backends/memory_mgmt.hh

```diff
@@ -0,0 +1,32 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+#include <memory>
+#include <sys/mman.h>
+
+namespace memory_mgmt{
+
+class FirstFitAllocator{
+public:
+struct MemBlock
+{
+    int size;
+    char used;
+    void *ptr;
+    std::shared_ptr<struct MemBlock> prev;
+    std::shared_ptr<struct MemBlock> next;
+};
+
+FirstFitAllocator(size_t sz);
+void *dev_malloc(size_t requested);
+void dev_free(void* block);
+void print_memory();
+
+
+void *device_memory = NULL;
+
+std::shared_ptr<struct MemBlock> heap_start;
+};
+
+}
```

Comment:
Sorry for being a bit pedantic but can you leave one new line at the end of the file?

Reply:
Must be present in latest commit